Sorting/HeapSort.cpp

```diff
@@ -0,0 +1,72 @@
+/*
+    Heap Sort :
+    Uses an array-implemented heap to sort list of numbers.
+
+    Time complexity:
+    O(nlog n), where n is the number of elements to be sorted
+
+    Space complexity:
+    O(1).
+*/
+
+#include <iostream>
+#include <vector>
+
+#include "SortingUtils.h"
+
+using namespace std;
+
+//creates ordered heap
+void heapify(vector<int> &heap, int parent, const int size, const int order, const bool showState){
+   int child = (parent*2)+1, temp;
+   while (child<size){//travels down children
+       displayState(heap);
+       if ((order*heap[child])<(order*heap[child+1])&&(child+1<size)){
+           child++;//child refers to the larger/smaller of the two children
+        }
+    else{}
+        if ((order*heap[parent])<(order*heap[child])){
+            swap(heap[parent],heap[child]);//if the parent is smaller(order==ascending) or larger(order==descending), then swap
+        }
+    parent = child;
+    child = (parent*2)+1;
+    displayState(heap);
+    }
+}
+//removes largest/smallest node, replaces it and reheapifies with new heap.
+void heapsort(vector<int> &heap, const int size, const int order, const bool showState){
+    int temp, last = size;
+    while (last>0){
+        swap(heap[0], heap[n-1]);
+        n-=1;
+        heapify(last,0,n, order, showState);
+    }

```

Comment:
You forgot to change `n` to `last` here. 

**Always compile and test your code before committing any changes.**